yoti-sdk-sandbox/src/main/java/com/yoti/api/client/sandbox/docs/request/check/SandboxCheckBuilder.java

```diff
@@ -0,0 +1,37 @@
+package com.yoti.api.client.sandbox.docs.request.check;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.yoti.api.client.sandbox.docs.request.check.report.SandboxBreakdown;
+import com.yoti.api.client.sandbox.docs.request.check.report.SandboxRecommendation;
+
+abstract class SandboxCheckBuilder<T extends SandboxCheckBuilder<T>> {
+
+    protected SandboxRecommendation recommendation;
+    protected List<SandboxBreakdown> breakdown;
+
+    public T withRecommendation(SandboxRecommendation recommendation) {
+        this.recommendation = recommendation;
+        return self();
+    }
+
+    public T withBreakdown(SandboxBreakdown breakdown) {
```

Comment:
I can do this in .NET for the "with" methods in this class:
```cs
public SandboxCheckBuilder<TCheck> WithBreakdown(SandboxBreakdown breakdown)
        {
            Breakdown.Add(breakdown);
            return this;
        }
```
Presume you can't do the same in Java to avoid having to use generics?

Reply:
(...although I do have a generic on the build() method)

Reply:
We don't have the generic as the check, rather the builder.  We need to know the type of the builder so we can return the correct child class from `with` methods (see `self()` abstract).

Reply:
Ah, I did actually think this was fine in .NET, but I've since realised that once you've used a parent `With`, you can't use the child `With`, so it's possible, but you'd have to order them. So will change that in .NET now

Reply:
Yep, that's why we have the builder as the generic and not the check itself 👍